### ISSUE TITLE
NullnessChecker false negative Path::getParent

### ISSUE THREAD

**Issue by xenoterracide** [typetools/checker-framework#4584]
```
Path::getParent
```

can return null if you are at `/`

```java
  @Test
  void test() {
    findConfig( Path.of( "" ) );
  }

  Optional<Path> findConfig( Path searchDir ) {
    var configFile = searchDir.resolve("foo.yml");

    if ( Files.exists( configFile ) ) {
      return Optional.of( configFile );
    }

    return this.findConfig( searchDir.getParent() );
  }
```

here's the relevant portion of my config, it's a `buildSrc` kotlin plugin but that shouldn't be relevant.

```kts
plugins {
  `java-library`
  `java-test-fixtures`
  id("brix.bom")
  id("net.ltgt.errorprone")
  id("org.checkerframework")
}

val checker = the<LibrariesForChecker>()

dependencies {
  errorprone("com.google.errorprone:error_prone_core:2.4.+")
  checkerFramework(checker.processor)
  compileOnly(checker.annotations)
  testFixturesCompileOnly(checker.annotations)
}

checkerFramework {
  excludeTests = true
  if (Files.exists(buildDir.toPath().resolve("../config/stubs"))) {
    extraJavacArgs.addAll(listOf("-Astubs=$buildDir/../config/stubs"))
  }
  checkers.addAll(
    listOf(
      "org.checkerframework.checker.nullness.NullnessChecker"
    )
  )
}
```

```
checkerFramework - A copy of JDK classes with Checker Framework type qualifiers inserted.
+--- org.checkerframework:checker:3.+ -> 3.12.0
|    +--- org.checkerframework:checker-qual:3.12.0
|    \--- org.checkerframework:checker-util:3.12.0
|         \--- org.checkerframework:checker-qual:3.12.0
+--- org.checkerframework:checker:{strictly 3.12.0} -> 3.12.0 (c)
+--- org.checkerframework:checker-qual:{strictly 3.12.0} -> 3.12.0 (c)
\--- org.checkerframework:checker-util:{strictly 3.12.0} -> 3.12.0 (c)
```

```
------------------------------------------------------------
Gradle 7.0
------------------------------------------------------------

Build time:   2021-04-09 22:27:31 UTC
Revision:     d5661e3f0e07a8caff705f1badf79fb5df8022c4

Kotlin:       1.4.31
Groovy:       3.0.7
Ant:          Apache Ant(TM) version 1.10.9 compiled on September 27 2020
JVM:          11.0.10 (AdoptOpenJDK 11.0.10+9)
OS:           Linux 5.11.10-200.fc33.x86_64 amd64
```

```
Fedora 33
5.11.10-200.fc33.x86_64
 09:47:23 up 26 days, 10:53,  1 user,  load average: 1.82, 1.38, 1.18
              total        used        free      shared  buff/cache   available
Mem:            15G         11G        1.3G        1.3G        3.4G        3.0G
Swap:           12G        2.6G        9.4G
```

https://scans.gradle.com/s/5tvngmmirwfva

**Comment by mernst** [comment]
You are right; thanks for the bug report.
Class `Path` has not yet been annotated for nullness.
I opened https://github.com/typetools/jdk/pull/109 with annotations for that class.

**Comment by mernst** [comment]
Fixed by https://github.com/typetools/jdk/commit/a4a0dc93ae32e431e41ecec64d82a09609209226
